### ISSUE TITLE
Dialect `WITH NO SCHEMA BINDING` in Redshift is unparsable

### ISSUE THREAD

**Issue by wakabame** [sqlfluff/sqlfluff#4806]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

`sqlfluff lint` fails on some Redshift View DDL including `WITH NO SHCHEMA BINDING`.

Syntax for `CREATE VIEW` is here: https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_VIEW.html

### Expected Behaviour

passes

### Observed Behaviour

```bash
$ cat create_view_with_noschema_binding.sql
CREATE VIEW test AS
SELECT user_id
FROM users
WITH NO SCHEMA BINDING;
$ 
$ sqlfluff lint create_view_with_noschema_binding.sql --dialect redshift
== [create_view_with_noschema_binding.sql] FAIL                                                                                                                                                                
L:   3 | P:  11 |  PRS | Line 3, Position 11: Found unparsable section: '\nWITH
                       | NO SCHEMA BINDING'
WARNING: Parsing errors found and dialect is set to 'redshift'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

1. Create SQL file `create_view_with_noschema_binding.sql` as below: 
    ```SQL
    CREATE VIEW test AS
    SELECT user_id
    FROM users
    WITH NO SCHEMA BINDING;
    ```
2. Apply `sqlfluff lint` with `--dialect redshift`
    ```shell
    sqlfluff lint create_view_with_noschema_binding.sql --dialect redshift
    ```

### Dialect

redshift

### Version

sqlfluff, version 2.0.5

### Configuration

nothing

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
